tests/ffn_model_tests.cpp

```diff
@@ -14,42 +14,183 @@
-  arma::mat input(448 * 448 * 3, 1), output;
-  input.ones();
 
   // Check output shape.
-  yolo.GetModel().Predict(input, output);
-  REQUIRE(output.n_cols == 1);
-  REQUIRE(output.n_rows == (7 * 7 * (5 * 2 + 20)));
+  ModelDimTest(yolo.GetModel(), input, (7 * 7 * (5 * 2 + 20)), 1);
+}
+
+/**
+ * Simple test for ResNet(18, 34, 50) models.
+ */
+TEST_CASE("ResNetModelTest", "[FFNModelsTests]")
+{
+  arma::mat input(224 * 224 * 3, 1);
+
+  // Check output shape for resnet18.
+  ResNet18 resnet18(3, 224, 224);
+  ModelDimTest(resnet18.GetModel(), input);
+
+  // Check output shape for resnet34.
+  ResNet34 resnet34(3, 224, 224);
+  ModelDimTest(resnet34.GetModel(), input);
+
+  // Check output shape for resnet50.
+  ResNet50 resnet50(3, 224, 224);
+  ModelDimTest(resnet50.GetModel(), input);
+}
+
+/**
+ * Simple test for ResNet101 models.
+ * Have been split from the ResNetModelTests because of memory requirements.
+ */
+TEST_CASE("ResNet101ModelTest", "[FFNModelsTests]")
+{
+  arma::mat input(224 * 224 * 3, 1);
+
+  // Check output shape for resnet101.
+  ResNet101 resnet101(3, 224, 224);
+  ModelDimTest(resnet101.GetModel(), input);
+}
+
+/**
+ * Simple test for ResNet152 models.
+ * Have been split from the ResNetModelTests because of memory requirements.
+ */
+TEST_CASE("ResNet152ModelTest", "[FFNModelsTests]")
+{
+  arma::mat input(224 * 224 * 3, 1);
+
+  // Check output shape for resnet152.
+  ResNet152 resnet152(3, 224, 224);
+  ModelDimTest(resnet152.GetModel(), input);
+}
+
+/**
+ * Test for pre-trained ResNet(18, 34, 50) models.
+ */
+TEST_CASE("PreTrainedResNetModelTest", "[FFNModelsTests]")
+{
+  arma::mat input(224 * 224 * 3, 1);
+
+  // Check output(referenced from PyTorch) for resnet18.
+  ResNet18 resnet18(3, 224, 224, true, true);
+  PreTrainedModelTest(resnet18.GetModel(), input, 0.00618362, 0.02469635);
+
+  // Check output(referenced from PyTorch) for resnet34.
+  ResNet34 resnet34(3, 224, 224, true, true);
+  PreTrainedModelTest(resnet34.GetModel(), input, 0.00664139, 0.02662659);
+
+  // Check output(referenced from PyTorch) for resnet50.
+  ResNet50 resnet50(3, 224, 224, true, true);
+  PreTrainedModelTest(resnet50.GetModel(), input, 0.00266838, 0.01067352);
+}
+
+/**
+ * Test for pre-trained ResNet101 model.
+ * Have been split from the PreTrainedResNetModelTests because of
+ *     memory requirements.
+ */
+TEST_CASE("PreTrainedResNet101ModelTest", "[FFNModelsTests]")
+{
+  arma::mat input(224 * 224 * 3, 1);
+
+  // Check output(referenced from PyTorch) for resnet101.
+  ResNet101 resnet101(3, 224, 224, true, true);
+  PreTrainedModelTest(resnet101.GetModel(), input, 0.00168228, 0.00670624);
+}
+
+/**
+ * Test for pre-trained ResNet152 model.
+ * Have been split from the PreTrainedResNetModelTests because of
+ *     memory requirements.
+ */
+TEST_CASE("PreTrainedResNetModel152Test", "[FFNModelsTests]")
+{
+  arma::mat input(224 * 224 * 3, 1);
+
+  // Check output for(referenced from PyTorch) resnet152.
```

Comment:
Can you add the link to the Pytorch code here as well.

Reply:
This was a local file, and these are just taken by having a input tensor of the same size as here and filling up with ones and passing that, basically the same procedure in torch.

Reply:
I see, thanks for the clarification.

Reply:
Sure, no problem.